gobblin-metrics/src/main/java/gobblin/metrics/event/sla/SlaEventSubmitter.java

```diff
@@ -0,0 +1,86 @@
+package gobblin.metrics.event.sla;
+
+import java.util.Map;
+import java.util.Properties;
+
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import gobblin.metrics.event.EventSubmitter;
+
+/**
+ * A wrapper around the {@link EventSubmitter} which can submit SLA Events.
+ */
+@Builder
+@Slf4j
+public class SlaEventSubmitter {
+
+  private EventSubmitter eventSubmitter;
+  private String eventName;
+  private String datasetUrn;
+  private String partition;
+  private String originTimestamp;
+  private String upstreamTimestamp;
+  private String recordCount;
+  private String previousPublishTimestamp;
+  private String dedupeStatus;
+  private String completenessPercentage;
+
+  public static SlaEventSubmitter of(String eventName, EventSubmitter eventSubmitter, Properties props) {
+    return SlaEventSubmitter.builder().eventName(eventName).eventSubmitter(eventSubmitter)
```

Comment:
I am trying to use lombok's standard builder. I have changed this static method to a constructor to match TimingEvent. I would like to avoid writing my own builder unless it is required. Let me know if you still think it is necessary, I will change it.